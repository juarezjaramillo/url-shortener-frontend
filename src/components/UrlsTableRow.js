import * as React from "react";

class UrlsTableRow extends React.Component {
    constructor(props) {
        super(props);
    }

    render() {
        const url = this.props.url;
        return (
            <tr>
                <td>{url.url}</td>
                <td>{url.title}</td>
                <td><a href={url.shortened_url} className="App-link">
                    {url.shortened_url}
                </a></td>
                <td>{url.hits}</td>
            </tr>
        );
    }
}

export default UrlsTableRow;