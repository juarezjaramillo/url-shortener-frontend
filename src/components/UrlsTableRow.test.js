import React from 'react';
import renderer from 'react-test-renderer';
import UrlsTableRow from "./UrlsTableRow";

it('renders correctly', () => {
    const fakeUrl = {
            id:1,
            url: "https://www.google.com",
            code: "ab",
            shortened_url: "https://test-url.com/v/ab",
            hits:2,
            title:"Google Test"
        }
    const tree = renderer
        .create(<UrlsTableRow url={fakeUrl} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});