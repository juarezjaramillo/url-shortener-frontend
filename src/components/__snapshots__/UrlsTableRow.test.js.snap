// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<tr>
  <td>
    https://www.google.com
  </td>
  <td>
    Google Test
  </td>
  <td>
    <a
      className="App-link"
      href="https://test-url.com/v/ab"
    >
      https://test-url.com/v/ab
    </a>
  </td>
  <td>
    2
  </td>
</tr>
`;
